### ISSUE TITLE
how to use the groundtruth data on the disk

### ISSUE THREAD

**Issue by KrisChou** [cchen156/Learning-to-See-in-the-Dark#25]
I don't have enough RAM，can you give me some suggestions to revise the code. Thanks for a lot.

**Comment by KevinJ-Huang** [comment]
I have the same problem,is it work by using the 16-bit groundtruth images processed by Rawpy?

**Comment by mbillouin2015** [comment]
I would also sincerely appreciate any advice you could give on how to revise the code in order to load the network using less RAM.

**Comment by oneTaken** [comment]
In my experiments, the RAM memory cost is ~110G. There are 1864 files during train phase(161 ground truth). The easiest way to reduce RAM cost is to reduce the training image number. And the second way is to crop the smaller center area as the input & gt. The third way is to random crop many patches to simulate the online cropping, so do not load everything into the memory.

**Comment by brouwa** [comment]
@oneTaken  How do you get these figures ?
I can train a network on a computer with 32Gb of RAM (+ mem swapping). Moreover, the original training indeed uses 161 ground truth pictures (i.e. 161 different scenes), each of them having at most 3 corresponding underexposed input images. Only the files including" _00_" in their name are considered.

**Comment by oneTaken** [comment]
@brouwa Do you mean the number **1864**? I use all the training pairs, not only the sequence id `_00_`, I start to load the training data by file `Sony_train_list`.

**Comment by brouwa** [comment]
@oneTaken  Yes I was referring to **1864**. That is indeed true if you use all the sequence ids, but this is not the default behaviour implemented in the published authors' code.

@KrisChou  Without even revising the code, you could try loading less images by simply deleting them in the dataset directory. If you remove (or rename, or move to another location) an `.ARW` file in the `dataset/long/` directory starting with `0`, then `glob` will no longer detect it, nor the corresponding underexposed images.
Of course, that means training on a smaller dataset...

**Comment by oneTaken** [comment]
@brouwa Yeah, the default use less data. Considering the limited datasets, I use all the training pairs. 
And what's your results with the smaller dataset, can you show the metrics to compare

**Comment by brouwa** [comment]
As stated by @cchen156 [here](https://github.com/cchen156/Learning-to-See-in-the-Dark/issues/22#issuecomment-396326334), the L1 loss on the training dataset converges to a little above 0.02. On the validation set, it is somewhere between 0.033 and 0.034.
Do you get improved results using on the whole training set ?

**Comment by brouwa** [comment]
I believe the PSNR and SSIM on the test set are the ones reported in the Table 3 of the paper : 28.88/0.787 for the Sony images. Do you get better results ?
I believe the dataset originally contains a batch of underexposed images for each scene and each ratio for the purpose of comparing SID to burst denoising methods.

**Comment by oneTaken** [comment]
@brouwa Do you calculate the PSNR & SSIM on the test set? I do not get better results. Yeah, the sequence may be used as burst comparison.

**Comment by himanshu112** [comment]
@oneTaken can you help me understand the scale image in the output set. The image seems to be very noisy.
